lib/auth/auth.go

```diff
@@ -2137,11 +2152,21 @@ func (a *Server) GenerateHostCerts(ctx context.Context, req *proto.HostCertsRequ
 		return nil, trace.Wrap(err)
 	}
 
+	var systemRoles []string
+	for _, r := range req.SystemRoles {
+		systemRoles = append(systemRoles, string(r))
+	}
+
+	if req.Role == types.RoleInstance && len(systemRoles) == 0 {
+		return nil, trace.BadParameter("cannot generate instance cert with no system roles")
+	}
+
 	// generate host TLS certificate
 	identity := tlsca.Identity{
 		Username:        HostFQDN(req.HostID, clusterName.GetClusterName()),
 		Groups:          []string{req.Role.String()},
```

Comment:
Yes, but more specifically, I'd like code that doesn't know what to do with a multirole cert to not fail in some odd or subtle/"almost right" way.  If a multirole cert can be misinterpreted as a single-role cert, that could result in confusing and difficult to diagnose behavior.  Much better IMO to avoid the potential ambiguity.